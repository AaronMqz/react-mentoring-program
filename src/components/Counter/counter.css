.counter-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.label-counter {
  font-size: 30px;
  font-weight: bold;
}

.button-counter {
  border-radius: 2px;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
  font-size: 16px;
  padding: 10px;
  margin: 10px;
  background: #f65261;
  width: 200px;
}
