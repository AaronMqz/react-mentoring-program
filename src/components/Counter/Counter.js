import React from "react";
import "./counter.css";

function Button(props) {
  return React.createElement(
    "div",
    { onClick: props.handleClick, className: "button-counter" },
    props.name
  );
}

class Counter extends React.Component {
  state = {
    count: this.props.initialValue,
  };

  handleIncrement = () => {
    this.setState({
      count: this.state.count + 1,
    });
  };

  handleDecrement = () => {
    this.setState({
      count: this.state.count - 1,
    });
  };

  render() {
    return React.createElement(
      "div",
      { className: "counter-container" },
      React.createElement(
        Button,
        { handleClick: this.handleIncrement, name: "Increment" },
        null
      ),
      React.createElement(
        Button,
        { handleClick: this.handleDecrement, name: "Decrement" },
        null
      ),
      React.createElement(
        "label",
        { className: "label-counter" },
        this.state.count
      )
    );
  }
}

export default Counter;
