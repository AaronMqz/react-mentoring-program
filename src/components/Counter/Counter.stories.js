import Counter from "../../components/Counter/Counter";

// More on how to set up stories at: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
export default {
  title: "Example/Counter",
  component: Counter,
  tags: ["autodocs"],
};

// More on writing stories with args: https://storybook.js.org/docs/7.0/react/writing-stories/args
export const Action = {
  args: {
    initialValue: 12,
  },
};
