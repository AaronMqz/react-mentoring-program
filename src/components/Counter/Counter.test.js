import React from "react";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import Counter from "./Counter";

afterEach(cleanup);

describe("Counter Tests:", () => {
  test("1. That component renders initial value provided in props", () => {
    render(<Counter initialValue={100} />);
    const result = screen.getByText(100);
    expect(result).toBeInTheDocument();
  });

  test("2. That a click event on 'decrement' button decrements the displayed value", () => {
    render(<Counter initialValue={100} />);
    const button = screen.getByText("Decrement");
    fireEvent.click(button);
    expect(screen.getByText(99)).toBeInTheDocument();
  });

  test("3. That a click event on 'increment' button decrements the displayed value", () => {
    render(<Counter initialValue={100} />);
    const button = screen.getByText("Increment");
    fireEvent.click(button);
    expect(screen.getByText(101)).toBeInTheDocument();
  });
});
