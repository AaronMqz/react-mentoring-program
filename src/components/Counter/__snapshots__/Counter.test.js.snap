// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Tests: 4. Snapshot renders correctly 1`] = `
<div
  className="counter-container"
>
  <div
    className="button-counter"
    onClick={[Function]}
  >
    Increment
  </div>
  <div
    className="button-counter"
    onClick={[Function]}
  >
    Decrement
  </div>
  <label
    className="label-counter"
  >
    100
  </label>
</div>
`;
